src/main/java/org/gridsuite/modification/server/NetworkModificationService.java

```diff
@@ -8,75 +8,85 @@
+
 /**
  * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
  */
 @ComponentScan(basePackageClasses = {NetworkStoreService.class})
 @Service
 class NetworkModificationService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkModificationService.class);
+
     @Autowired
     private NetworkStoreService networkStoreService;
 
-    private Network getNetwork(UUID networkUuid) {
-        try {
-            return networkStoreService.getNetwork(networkUuid);
-        } catch (PowsyblException e) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Network '" + networkUuid + "' not found");
-        }
+    private Set<String> doModification(Network network, Runnable modification) {
+        return doModification(network, modification, MODIFICATION_ERROR);
     }
 
-    public Set<String> changeSwitchState(UUID networkUuid, String switchId, boolean open) {
-        Network network = getNetwork(networkUuid);
-
-        Switch sw = network.getSwitch(switchId);
-        if (sw == null) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Switch " + switchId + " not found");
-        }
-
-        if (sw.isOpen() != open) {
-            sw.setOpen(open);
+    private Set<String> doModification(Network network, Runnable modification, NetworkModificationException.Type typeIfError) {
+        try {
+            DefaultNetworkStoreListener listener = new DefaultNetworkStoreListener();
+            network.addListener(listener);
+            modification.run();
 
             networkStoreService.flush(network);
 
-            return Set.of(sw.getVoltageLevel().getSubstation().getId());
-        } else {
-            return Set.of();
+            return listener.getModifications();
+        } catch (Exception e) {
+            NetworkModificationException exc = new NetworkModificationException(typeIfError, e.getMessage());
+            LOGGER.error(exc.getMessage());
+            throw exc;
         }
     }
 
-    public GroovyScriptResult applyGroovyScript(UUID networkUuid, String groovyScript) {
-        CompilerConfiguration conf = new CompilerConfiguration();
-        Network network = getNetwork(networkUuid);
-        DefaultNetworkStoreListener listener = new DefaultNetworkStoreListener();
-        network.addListener(listener);
-
-        Binding binding = new Binding();
-        binding.setProperty("network", network);
-        GroovyShell shell = new GroovyShell(binding, conf);
+    private Mono<Network> getNetwork(UUID networkUuid) {
         try {
-            shell.evaluate(groovyScript);
-            networkStoreService.flush(network);
-
-            return new GroovyScriptResult(true, listener.getModifications());
-        } catch (Exception ignored) {
-            return new GroovyScriptResult(false, listener.getModifications());
+            return Mono.just(networkStoreService.getNetwork(networkUuid));
```

Comment:
use a thread pool to perform this blocking work ?

Reply:
Done